server/lib/chat/chat-service.ts

```diff
@@ -765,18 +840,37 @@ export default class ChatService {
     return userSockets
   }
 
-  private subscribeUserToChannel(
-    userSockets: UserSocketsGroup,
-    channelInfo: ChannelInfo,
-    channelPermissions: ChannelPermissions,
-  ) {
-    userSockets.subscribe<ChatInitEvent>(getChannelPath(channelInfo.id), () => ({
-      action: 'init3',
-      channelInfo,
-      activeUserIds: this.state.channels.get(channelInfo.id)!.toArray(),
-      selfPermissions: channelPermissions,
-    }))
-    userSockets.subscribe(getChannelUserPath(channelInfo.id, userSockets.userId))
+  private subscribeUserToChannel(userSockets: UserSocketsGroup, channelId: SbChannelId) {
+    userSockets.subscribe<ChatInitEvent>(getChannelPath(channelId), async () => {
+      try {
+        const [channelInfo, userChannelEntry] = await Promise.all([
+          getChannelInfo(channelId),
+          getUserChannelEntryForUser(userSockets.userId, channelId),
+        ])
+        if (!channelInfo) {
+          throw new ChatServiceError(ChatServiceErrorCode.ChannelNotFound, 'Channel not found')
```

Comment:
I'm not sure why we're still throwing `ChatServiceErrors` here like they're going to be meaningful in any way, the only thing handling this is the `catch` below, and it does not care about `ChatServiceError` in any way